dropwizard-jdbi3/src/main/java/io/dropwizard/jdbi3/JdbiFactory.java

```diff
@@ -0,0 +1,118 @@
+package io.dropwizard.jdbi3;
+
+import com.codahale.metrics.jdbi3.InstrumentedTimingCollector;
+import com.codahale.metrics.jdbi3.strategies.SmartNameStrategy;
+import com.codahale.metrics.jdbi3.strategies.StatementNameStrategy;
+import io.dropwizard.db.ManagedDataSource;
+import io.dropwizard.db.PooledDataSourceFactory;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.util.Duration;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.SqlStatements;
+import org.jdbi.v3.core.statement.TemplateEngine;
+
+public class JdbiFactory {
+    private final StatementNameStrategy nameStrategy;
+
+    public JdbiFactory() {
+        this(new SmartNameStrategy());
+    }
+
+    public JdbiFactory(StatementNameStrategy nameStrategy) {
+        this.nameStrategy = nameStrategy;
+    }
+
+    /**
+     * Build a fully configured {@link Jdbi} instance managed by the DropWizard lifecycle
+     * with the configured health check; this method should not be overridden
+     * (instead, override {@link #newInstance(ManagedDataSource)} and
+     * {@link #configure(Jdbi, PooledDataSourceFactory)})
+     *
+     * @param environment
+     * @param configuration
+     * @param name
+     * @return A fully configured {@link Jdbi} object using a managed data source
+     * based on the specified environment and configuration
+     * @see #build(Environment, PooledDataSourceFactory, ManagedDataSource,
+     * String)
+     */
+    public Jdbi build(Environment environment,
+                      PooledDataSourceFactory configuration,
+                      String name) {
+        final ManagedDataSource dataSource = configuration.build(environment.metrics(), name);
+        return build(environment, configuration, dataSource, name);
+    }
+
+    /**
+     * Build a fully configured {@link Jdbi} instance managed by the DropWizard lifecycle
+     * with the configured health check; this method should not be overridden
+     * (instead, override {@link #newInstance(ManagedDataSource)} and
+     * {@link #configure(Jdbi, PooledDataSourceFactory)})
+     *
+     * @param environment
+     * @param configuration
+     * @param dataSource
+     * @param name
+     * @return A fully configured {@link Jdbi} object
+     */
+    public Jdbi build(Environment environment,
+                      PooledDataSourceFactory configuration,
+                      ManagedDataSource dataSource,
+                      String name) {
+
+        // Create the instance
+        final Jdbi jdbi = this.newInstance(dataSource);
```

Comment:
`this` seems to be redundant, because there's no superclass for `JdbiFactory` which has a method with the same name.

Reply:
Removed 'this'